coral-service/src/main/java/com/linkedin/coral/coralservice/utils/CoralUtils.java

```diff
@@ -0,0 +1,87 @@
+package com.linkedin.coral.coralservice.utils;
+
+import com.linkedin.coral.common.HiveMetastoreClient;
+import com.linkedin.coral.common.HiveMscAdapter;
+import com.linkedin.coral.coralservice.metastore.MetastoreProvider;
+import com.linkedin.coral.hive.hive2rel.HiveToRelConverter;
+import com.linkedin.coral.trino.rel2trino.RelToTrinoConverter;
+import com.linkedin.coral.trino.trino2rel.TrinoToRelConverter;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.UUID;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.metastore.api.MetaException;
+import org.apache.hadoop.hive.ql.CommandNeedRetryException;
+import org.apache.hadoop.hive.ql.Driver;
+import org.apache.hadoop.hive.ql.metadata.Hive;
+import org.apache.hadoop.hive.ql.metadata.HiveException;
+import org.apache.hadoop.hive.ql.processors.CommandProcessorResponse;
+import org.apache.hadoop.hive.ql.session.SessionState;
+
+/**
+ * Utility class to provide Coral functionality to Coral Service.
+ */
+public class CoralUtils {
+
+  public static HiveMetastoreClient hiveMetastoreClient;
+
+  // Converters
+  public static HiveToRelConverter hiveToRelConverter;
+  public static TrinoToRelConverter trinoToRelConverter;
+  public static RelToTrinoConverter relToTrinoConverter = new RelToTrinoConverter();
+
+  // Local Metastore
+  public static Driver driver;
+  public static HiveConf conf;
+  public static final String CORAL_SERVICE_DIR = "coral.service.test.dir";
+
+  public static void initHiveMetastoreClient() throws Exception{
+    // Connect to remote production Hive Metastore Client
+    hiveMetastoreClient  = MetastoreProvider.getMetastoreClient();
+    hiveToRelConverter = new HiveToRelConverter(hiveMetastoreClient);
+    trinoToRelConverter  = new TrinoToRelConverter(hiveMetastoreClient);
+  }
+
+  public static void initLocalMetastore() throws IOException, HiveException, MetaException {
+    // Create a temporary local metastore
+    conf = loadResourceHiveConf();
+    String tempDir = conf.get(CORAL_SERVICE_DIR);
+    System.out.println("Temp Workspace: " + tempDir);
+    FileUtils.deleteDirectory(new File(tempDir));
+    SessionState.start(conf);
+    driver = new Driver(conf);
+
+    hiveMetastoreClient = new HiveMscAdapter(Hive.get(conf).getMSC());
+    hiveToRelConverter = new HiveToRelConverter(hiveMetastoreClient);
+    trinoToRelConverter  = new TrinoToRelConverter(hiveMetastoreClient);
+  }
+
+  public static void run(Driver driver, String sql) {
+    while (true) {
+      try {
+        CommandProcessorResponse result = driver.run(sql);
+        if (result.getException() != null) {
+          throw new RuntimeException("Execution failed for: " + sql, result.getException());
+        }
+      } catch (CommandNeedRetryException e) {
+        continue;
+      }
+      break;
+    }
+  }
+
+  public static HiveConf loadResourceHiveConf() {
```

Comment:
Provide this as a `bean` instead of a static function. Dependency injection is the most powerful feature of springboot.